build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/task/helper/TaskHelperPublications.java

```diff
@@ -150,19 +152,26 @@ public void checkDependsOnArtifactsToPublish() {
         // the 'generate Pom file' task
         for (MavenPublication mavenPublication : mavenPublications) {
             if (!(mavenPublication instanceof MavenPublicationInternal)) {
-                // TODO: Check how the output files can be extracted without using getPublishableFiles
                 log.warn("Maven publication name '{}' is of unsupported type '{}'!",
                     mavenPublication.getName(), mavenPublication.getClass());
                 continue;
             }
-            dependsOn(((MavenPublicationInternal) mavenPublication).getPublishableFiles());
+            dependOn(mavenPublication);
             String capitalizedPublicationName = mavenPublication.getName().substring(0, 1).toUpperCase() +
                 mavenPublication.getName().substring(1);
             dependsOn(String.format("%s:generatePomFileFor%sPublication",
                 getProject().getPath(), capitalizedPublicationName));
         }
     }
 
+    private void dependOn(Publication publication) {
+        // TODO: Check how we can find the artifact dependencies without using internal api's.
+	// Based on org.gradle.plugins.signing.Sign#sign
+	PublicationInternal<?> publicationInternal = (PublicationInternal<?>) publication;
+	dependsOn((Callable<Set<? extends PublicationArtifact>>) publicationInternal::getPublishableArtifacts);
+	publicationInternal.allPublishableArtifacts(this::dependsOn);
```

Comment:
@steven-aerts, are you sure this line is needed?
The previous line already added the whole collection.

Reply:
I agree with you, I think it might be possible to remove it.

As mentioned above for this code I based myself on Sigh#sign, which also does those [two steps](https://github.com/gradle/gradle/blob/master/subprojects/signing/src/main/java/org/gradle/plugins/signing/Sign.java#L176).

The second step is probably not needed, but kept it to be sure/the same.